x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.scss

```diff
@@ -1,34 +1,45 @@
 @import '../../../mixins';
 
 .lnsLayerPanel {
-  margin-bottom: $euiSizeS;
+  margin-bottom: $euiSize;
 
   // disable focus ring for mouse clicks, leave it for keyboard users
   &:focus:not(:focus-visible) {
     animation: none !important; // sass-lint:disable-line no-important
   }
 }
 
-.lnsLayerPanel__sourceFlexItem {
-  max-width: calc(100% - #{$euiSize * 3.625});
+.lnsLayerPanel__layerHeader {
+  padding: $euiSize;
+  border-bottom: $euiBorderThin;
+}
+
+// fixes truncation for too long chart switcher labels
+.lnsLayerPanel__layerSettingsWrapper {
+  min-width: 0;
 }
 
-.lnsLayerPanel__settingsFlexItem:empty + .lnsLayerPanel__sourceFlexItem {
-  max-width: calc(100% - #{$euiSizeS});
+.lnsLayerPanel__settingsStaticHeaderIcon {
+  margin-right: $euiSizeS;
+  vertical-align: inherit;
 }
 
-.lnsLayerPanel__settingsFlexItem:empty {
-  margin: 0;
+.lnsLayerPanel__settingsStaticHeader {
+  display: inline;
 }
```

Comment:
If reverting to using an `EuiFlexGroup` structure as suggested in my previous comment, I think that will render these styles unnecessary. Consider removing if so.